### PR TITLE
DBZ-8318 Explain how to manage SPI and bundled jars for debezium engine applications

### PR DESCRIPTION
**Pull request by mfvitale**
closes: https://issues.redhat.com/browse/DBZ-8318

### CODE REVIEW

**Comment by mfvitale** [comment]
PR for the FAQ https://github.com/debezium/debezium.github.io/pull/1075

File: documentation/modules/ROOT/pages/development/engine.adoc [hunk 1]
```diff
@@ -82,6 +82,30 @@ Or for the MongoDB connector:
...
 Other connectors are used in a similar manner, except with connector-specific configuration, topics, and events.
 
+=== Package you project
+Debezium uses the SPI via the ServiceLoader to load different implementations depending on the connector type or for custom implementations.
+
+When an interface has multiple implementations — for instance, `io.debezium.snapshot.spi.SnapshotLock`, which has a default implementation in the core and specific ones for each connector — you must explicitly configure your build tools to merge the `META-INF/services files`.
+If this step is missed, Debezium may fail to locate the required implementation.
+
+To address this, instruct your build tools accordingly. For example, if you are using the https://maven.apache.org/plugins/maven-shade-plugin[Maven shade plugin],
+you need to add the https://maven.apache.org/plugins/maven-shade-plugin/examples/resource-transformers.html#ServicesResourceTransformer[`ServicesResourceTransformer`] transformer:
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/development/engine.adoc:92]:
```suggestion
For example, if you are using the https://maven.apache.org/plugins/maven-shade-plugin[Maven shade plugin],
add the https://maven.apache.org/plugins/maven-shade-plugin/examples/resource-transformers.html#ServicesResourceTransformer[`ServicesResourceTransformer`] transformer, as shown in the following example:
```

File: documentation/modules/ROOT/pages/development/engine.adoc [hunk 2]
```diff
@@ -82,6 +82,30 @@ Or for the MongoDB connector:
...
+ <transformers>
+    ...
+    <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer" />
+    ...
+ </transformers>
+...
+</configuration>
+----
+
+Alternatively, if using the https://maven.apache.org/plugins/maven-assembly-plugin/index.html[Maven Assembly plug-in] you can utilize the https://maven.apache.org/plugins/maven-assembly-plugin/examples/single/using-container-descriptor-handlers.html[metaInf-services container descriptor handlers].
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/development/engine.adoc:107]:
```suggestion
Alternatively, if you use the https://maven.apache.org/plugins/maven-assembly-plugin/index.html[Maven Assembly plug-in], you can use the https://maven.apache.org/plugins/maven-assembly-plugin/examples/single/using-container-descriptor-handlers.html[metaInf-services container descriptor handlers].
```

**Review by roldanbob — CHANGES_REQUESTED** [review]
@mfvitale @jpechane I made a few suggestions. Let me know if you have any questions. Thanks!

**Comment by mfvitale** [comment]
> @mfvitale @jpechane I made a few suggestions. Let me know if you have any questions. Thanks!

@roldanbob Thanks for your review. I have addressed your suggestions.

**Review by roldanbob — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**